### PR TITLE
Feature/85 choose an activity key

### PR DESCRIPTION
**Pull request by nils-schmitt**
This PR will allow for the key after which the xes and dfg is structured to be selectable. 
I modified the handling of the dataframe in `views.py`, especially the confusing way of back and forth formatting of column names and the way of accessing the dataframe. Additionally, I removed the need for any `.csv` file and I fixed the bug, that the result view crashed when at any point the filters where so selective that no dfg is resulting and afterwards selecting again enough to show a dfg. (I can elaborate on this in the next meeting.)

### CODE REVIEW

**Comment by nils-schmitt** [comment]
In `views.py` I use a switch-case (or match-case in python). Unfortunately it was integrated in 3.10, which is why the other pylints fail. Should I change it to a less readable if-elif-else structure or do we ignore 3.8 and 3.9 starting now?

**Comment by tkv29** [comment]
@nils-schmitt forget for now the first point, I will write to this later sth

**Comment by nils-schmitt** [comment]
Weird, does work on both of my machines... Try to pull again?

**Comment by tkv29** [comment]
@nils-schmitt based on the pr description, you did #98? 

cc: @FR-SON

**Comment by tkv29** [comment]
regarding the web scraper, do I need to anything else except installing selenium and the webdriver?
I got this error:
image

**Comment by nils-schmitt** [comment]
You might need to install Firefox. As I wrote in the script, it is Firefox exclusive. Could be changed to being chrome exclusive, but to get an idea of the functionality i thought, this is enough

**Comment by tkv29** [comment]
I also have firefox installed but got the erros above. Maybe it makes sense to specify a path where the firefox file is? Seems like it cant find my firefox

The radio buttons looks good!

**Comment by nils-schmitt** [comment]
I think, that the errors come from you running the script off ubuntu. As the geckodriver has to be installed you might need a sudo

**Comment by nils-schmitt** [comment]
@tkv29 Do you want me to implement further changes? If not, please request reviewers.

**Comment by tkv29** [comment]
Functionality looks good for me! GJ! Web Scraping will be outsourced in another PR

File: tracex_project/tracex/logs/execution_time.log
Comment (PitButtchereit) [inline tracex_project/tracex/logs/execution_time.log:1]: This file should not be deleted

File: tracex_project/tracex/logs/tokens_used.log
Comment (PitButtchereit) [inline tracex_project/tracex/logs/tokens_used.log:1]: This file should not be deleted

File: tracex_project/trace_comparator/logs/execution_time.log
Comment (PitButtchereit) [inline tracex_project/trace_comparator/logs/execution_time.log:1]: I thought all gpt requests would be logged in `tracex/logs`

File: tracex_project/patient_journey_generator/logs/execution_time.log
Comment (PitButtchereit) [inline tracex_project/patient_journey_generator/logs/execution_time.log:1]: I thought all gpt requests would be logged in `tracex/logs`

File: tracex_project/extraction/views.py [hunk 1]
```diff
@@ -114,25 +105,37 @@ def get_context_data(self, **kwargs):
...
+                    "attribute_location": orchestrator.get_configuration().locations,
+                    "concept:name": orchestrator.get_configuration().event_types,
+                }
+            case "attribute_location":
+                filter_dict = {
+                    "concept:name": orchestrator.get_configuration().locations,
+                    "event_type": orchestrator.get_configuration().event_types,
+                }
+            case _:
+                filter_dict = {}
```
Comment (PitButtchereit) [inline tracex_project/extraction/views.py:129]: Can this also be outsourced to the Orchestrator to a method like `get_activity_key`? For me it is a bit weird that we have this condition in the views even though it does not directly relate to something getting displayed

**Review by PitButtchereit — CHANGES_REQUESTED** [review]

**Review by FR-SON — APPROVED** [review]

File: tracex_project/extraction/views.py (hunk 1 above)
Comment (nils-schmitt) [reply tracex_project/extraction/views.py:129]: Will be tackled in refactoring-2

**Review by PitButtchereit — APPROVED** [review]

**Pull request merged**